CMakeLists.txt

```diff
@@ -25,6 +28,8 @@ set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 include(CTest)
 enable_testing()
 
+include(GNUInstallDirs)
```

Comment:
What is this? Does it work also in Windows?

Reply:
yes and no 😄 it's just a generic standard CMake configuration file that just so happens to be developed by GNU. What it does is set a lot of CMake definitions and variables for installation, mainly for Linux (but with platform compatibility). E.g. shared libraries should have their RPATH set if that property exists (which means: Unix). See also the documentation in https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html

For the Windows shared libraries, the user dynamically linking to this DLL, the `install(IMPORTED_RUNTIME_ARTIFACTS library)` already does the job

Reply:
NOTE: for Unix, the user also still needs to set `set_target_properties(<target> PROPERTIES INSTALL_RPATH_USE_LINK_PATH TRUE)` to ensure that the properties propagate accordingly